app/src/main/java/com/example/sudoku/Play.kt

```diff
@@ -71,6 +68,22 @@ class Play : Fragment() {
         gameDisplayContext = GameDisplayContext(game.current, game.solution)
         sudokuBoardGridAdapter = SudokuBoardGridAdapter(this.requireContext(), gameDisplayContext)
         sudokuGridView.adapter = sudokuBoardGridAdapter
+
+        game.score().observe(viewLifecycleOwner, {
```

Comment:
This sets a callback to be used every time when the score is updated, right? Looks easier than what we had before.